src/adapter.ts

```diff
@@ -395,17 +377,20 @@ export class MongooseAdapter {
       const policyRuleAST = model.model.get('p') instanceof Map ? model.model.get('p') : new Map();
       const groupingPolicyAST = model.model.get('g') instanceof Map ? model.model.get('g') : new Map();
 
-      // @ts-ignore
-      for (const [ptype, ast] of policyRuleAST) {
-        for (const rule of ast.policy) {
-          lines.push(this.savePolicyLine(ptype, rule));
+      if (policyRuleAST) {
```

Comment:
Hmm. When does this apply? [Line 377](https://github.com/node-casbin/mongoose-adapter/pull/47/files#diff-90f5592a8a85ef6ea1f847351c4ec2ad2f646d34f426d1e9feb2ed7411f344ccR377) should ensure that it's at least a map every time.

Reply:
Typescript also create a warning on this. This can make it happy. :(

Reply:
Please change it to this then:

```typesript
const policyRuleAST = model.model.get('p') instanceof Map ? model.model.get('p')! : new Map();
const groupingPolicyAST = model.model.get('g') instanceof Map ? model.model.get('g')! : new Map();
```

We should tell the compiler that model cannot be `undefined` and not add unnecessary if-statements. (Which   instance check should do, but I am not sure if there is more elegant way of doing this in typescript)

Reply:
@Zxilly